src/daemon/auth/providers/openid_connect/provider.rs

```diff
@@ -338,10 +341,24 @@
                                         }
                                     }
                                     Err(err) => {
-                                        warn!(
-                                            "OpenID Connect: Could not extend login session for user '{}': {}",
-                                            &session.id, err
-                                        );
+                                        match &err {
+                                            // this is where the RFC-6749 5.2 Error Response is received and
+                                            // return to the caller. It's the responsibility of the caller
+                                            // to decide whether to retry or report back to the user.
```

Comment:
Could this be moved up into the Err() match above thus avoiding the need for an extra match?

Reply:
This is what I am still working on, there's other variants in here that need to be handled, I think. If this turns out to be only usefull variant, I'll collapse this one into its parent.